src/utils/deprecated/deprecated.tsx

```diff
@@ -5,14 +5,34 @@
 
 import React, { useEffect } from 'react';
 
+interface IDeprecatedComponentWarning {
+  NewComponent: React.ComponentType<any>;
+  version: string;
+}
+
+interface IDeprecatedPropWarning {
+  deprecatedProp: any;
+  deprecatedPropName: string;
+  version: string;
+}
+
 export const getDeprecatedMessage = (message: string): string =>
   `[DEPRECATED] ${message}`;
 
-export const deprecated = (message: string) => {
+export const deprecatedComponentWarning = ({
+  NewComponent,
+  version,
+}: IDeprecatedComponentWarning) => {
   return <T extends React.ComponentType<any>>(Component: T): T => {
+    const deprecatedComponentName = Component.displayName || Component.name;
+    const newComponentName = NewComponent.displayName || NewComponent.name;
+
     const DeprecatedWrapper = (props: React.ComponentProps<T>) => {
       useEffect(() => {
-        console.warn(getDeprecatedMessage(message));
+        const formattedMessage = `${deprecatedComponentName} is deprecated in favor of ${newComponentName} and will be removed in v${version}.`;
+        const deprecatedMessage = getDeprecatedMessage(formattedMessage);
```

Comment:
It may be nice to also pass in a custom deprecation message. For example, we're thinking about deprecating the current `OuiCodeEditor` in favor of `@osd/monaco`, which might require a different message than this allows

Reply:
Good suggestion, now we can pass the `getMessage` function